core/src/block.rs

```diff
@@ -620,68 +631,61 @@ fn revalidate_transactions(
     is_genesis: bool,
 ) -> Result<(), TransactionRevalidationError> {
     // Check that valid transactions are still valid
-    let _transactions = transactions
-        .iter()
-        .cloned()
-        .map(|tx| {
-            if is_genesis {
-                Ok(AcceptedTransaction::accept_genesis(GenesisTransaction(tx)))
+    let mut tx_iter = transactions.iter().cloned().peekable();
+    let mut rejected_tx_iter = rejected_transactions.iter().cloned().peekable();
+    for tx_index in 0..u64::MAX {
```

Comment:
Oh no. I'm sure there is a way to not to iterate over such a big amount of numbers. There should be a way to place exit condition in the loop condition here